.github/workflows/agw-build-publish-container.yml

```diff
@@ -9,100 +9,118 @@
+      - run: echo "docker-builder-python conclusion = ${{ steps.docker-builder-python.conclusion }}"
 
       - uses: ./.github/workflows/composite/docker-builder-agw
+        id: docker-builder-go
         with:
           REGISTRY_USERNAME: ${{ secrets.JFROG_USERNAME }}
           REGISTRY_PASSWORD: ${{ secrets.JFROG_PASSWORD }}
-          REGISTRY: ${{ env.DOCKER_REGISTRY }}
-          IMAGE: ${{ matrix.DOCKER_IMAGE }}
-          FILE: ${{ matrix.DOCKER_FILE }}
+          REGISTRY: ${{ env.registry }}
+          FILE: feg/gateway/docker/go/Dockerfile
+          TAGS: ${{ env.registry }}/gateway_go:${{ steps.get-short-git-sha.outputs.sha }}
+      - run: echo "Go container image digest is ${{ steps.docker-builder-go.outputs.digest }}"
+      - run: echo "docker-builder-go conclusion = ${{ steps.docker-builder-go.conclusion }}"
 
   build-containers-ghz:
     runs-on: ubuntu-latest
     needs: build-containers
-    env:
-      DOCKER_REGISTRY: agw-ci.artifactory.magmacore.org
-    strategy:
-      fail-fast: false
-      matrix:
-        include:
-          - DOCKER_IMAGE: ghz_gateway_python
-            DOCKER_CONTEXT: lte/gateway/docker/ghz
-          - DOCKER_IMAGE: ghz_gateway_c
-            DOCKER_CONTEXT: lte/gateway/docker/ghz
-
     steps:
-      - run: echo "DOCKER_REGISTRY=agw-test.artifactory.magmacore.org" >> $GITHUB_ENV
-        if: github.ref == 'refs/heads/master'
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin@v3.1.0
 
-      - run: echo "Publishing images to ${{ env.DOCKER_REGISTRY }}"
+      - id: get-short-git-sha
+        run: echo sha=${GITHUB_SHA:0:8} >> $GITHUB_OUTPUT
 
-      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin@v3.1.0
+      - uses: ./.github/workflows/composite/docker-builder-agw
+        with:
+          REGISTRY_USERNAME: ${{ secrets.JFROG_USERNAME }}
+          REGISTRY_PASSWORD: ${{ secrets.JFROG_PASSWORD }}
+          REGISTRY: ${{ env.registry }}
+          TAGS: ${{ env.registry }}/ghz_gateway_c:${{ steps.get-short-git-sha.outputs.sha }}
+          CONTEXT: lte/gateway/docker/ghz
 
       - uses: ./.github/workflows/composite/docker-builder-agw
         with:
           REGISTRY_USERNAME: ${{ secrets.JFROG_USERNAME }}
           REGISTRY_PASSWORD: ${{ secrets.JFROG_PASSWORD }}
-          REGISTRY: ${{ env.DOCKER_REGISTRY }}
-          IMAGE: ${{ matrix.DOCKER_IMAGE }}
-          CONTEXT: ${{ matrix.DOCKER_CONTEXT }}
+          REGISTRY: ${{ env.registry }}
+          TAGS: ${{ env.registry }}/ghz_gateway_python:${{ steps.get-short-git-sha.outputs.sha }}
+          CONTEXT: lte/gateway/docker/ghz
+
+  test-containers-precommit:
+    needs: build-containers
+    # We only want to trigger the tests if the build-containers job uploaded the images.
+    # The following condition is a crude heuristic for this limitation.
+    if: github.event_name == 'push' || github.event_name == 'workflow_dispatch'
```

Comment:
Why would `workflow_dispatch` guarantee an upload? Would it make more sense to check if the repository owner is magma, or just if the secret for uploading exists?

Reply:
I think you are right and checking for the secrets is better. I think with `workflow_dispatch` and `push` we know that the secrets are there (as opposed to `pull_request`), so that is just an indirect way of expressing that. I'll change that to check for the secrets.

Reply:
You can have `workflow_dispatch` and `push` on forks :wink:

Reply:
Wait - you cannot use secrets in Github Action if conditions...

Reply:
I would leave with this workaround for now, we can still improve upon this later.

Reply:
In `bazel.yml` we have the workaround that the bash script takes a secret as optional argument which exists or doesn't, instead of a direct `if` condition.

Reply:
Here we could at least check if `github.repository_owner == 'magma'`

Reply:
For testing it is helpful to be able to run this on forks, I did this a lot while working on this.